xshop/core/tests/test_api.py

```diff
@@ -0,0 +1,112 @@
+from django.urls import reverse
+from rest_framework.test import APITestCase, APIClient
+from model_bakery import baker
+from django.contrib.auth import get_user_model
+
+from ..models import Product
+
+User = get_user_model()
+
+
+def product_detail_url(product_id):
+    return reverse("core_api:product_detail_patch", args=[product_id])
+
+
+class ProductCrudOperationsTest(APITestCase):
+    def setUp(self) -> None:
+        self.user = baker.make(User, mobile="01010092181", name="Ahmed Loay Shahwan",)
+        self.user.save()
+        self.product1 = baker.make(
+            Product, name="Prod1", stock=15, price=12, added_by=self.user
+        )
+        self.product2 = baker.make(
+            Product, name="Prod2", stock=10, price=41, added_by=self.user
+        )
+        self.product1.save()
+        self.product2.save()
+        self.client = APIClient()
+        self.url = reverse("core_api:product_list_create")
+
+    def test_get_all_products(self):
+        resp = self.client.get(self.url)
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(len(resp.data), 2)
+
+    def test_create_valid_product_with_price(self):
+        product = {
+            "name": "test",
+            "stock": 14,
+            "price": "17.00",
+            "added_by": self.user.id,
+        }
+        resp = self.client.post(self.url, product)
+        self.assertEqual(resp.status_code, 200)
```

Comment:
```Python
from rest_framework import status

status.HTTP_200_OK  # instead of just 200
status.HTTP_400_BAD_REQUEST  # instead of just 400
status.HTTP_404_NOT_FOUND  # instead of just 404
```

Reply:
replace all 200,400,404 with the corresponding status.* as mentioned above